### PR TITLE
ci: add splunk update semver

### PR DESCRIPTION
**Pull request by mbruzda-splunk**
Changes update-semver action from haya14busa to splunk owned one

JIRA: https://splunk.atlassian.net/browse/ADDON-68596

### CODE REVIEW

**Comment by mkolasinski-splunk** [comment]
@mbruzda-splunk please attach test run link

**Comment by mbruzda-splunk** [comment]
This needs to be backported to main

**Comment by mbruzda-splunk** [comment]
@mkolasinski-splunk this changes build-test-release, not reusable workflow, thus test run won't be provided

File: .github/workflows/build-test-release.yaml [hunk 1]
```diff
@@ -58,9 +58,14 @@ jobs:
   update-semver:
     if: startsWith(github.ref, 'refs/tags/v')
     needs: publish
-    permissions:
-      contents: write
```
Comment (artemrys) [inline .github/workflows/build-test-release.yaml:62]: Why do we remove these lines?

Comment (mbruzda-splunk) [reply .github/workflows/build-test-release.yaml:62]: We have global permissions in this case

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**